scipy/stats/_qmc.py

```diff
@@ -898,6 +1044,63 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
         self.num_generated += n
         return samples
 
+    def _random_optimal(self, n: IntNumber = 1) -> np.ndarray:
+        """Draw `n` in the half-open interval ``[0, 1)``.
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate in the parameter space. Default is 1.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            LHS sample.
+
+        """
+        if self.d == 0 or n == 0:
+            return np.empty((n, self.d))
+
+        best_sample = self._random(n=n)
+        best_disc = discrepancy(best_sample)
+
+        if n == 1:
+            return best_sample
+
+        bounds = ([0, self.d - 1],
+                  [0, n - 1],
+                  [0, n - 1])
+
+        n_nochange = 0
+        n_iters = 0
+        while "Optimization loop":
```

Comment:
Maybe refactor this in terms of `while n_nochange < 10 and n_iters < 10000` to avoid the appearance of an infinite loop and remove some lines below.